src/CommonLib/hostfxr_utility.cpp

```diff
@@ -276,14 +277,22 @@ HOSTFXR_UTILITY::SetHostFxrArguments(
     INT         argc = 0;
     PCWSTR*     argv = NULL;
     LPWSTR*     pwzArgs = NULL;
+    STRU        struTempPath;
 
     pwzArgs = CommandLineToArgvW(struArgumentsFromConfig->QueryStr(), &argc);
 
-    if (pwzArgs == NULL)
+    if (pwzArgs == NULL || argc < 1)
     {
         goto Finished;
     }
 
+    // Try to convert the application dll from a relative to an absolute path
+    // Don't record this failure as pwzArgs[0] may already be an absolute path to the dll.
+    if (!FAILED(UTILITY::ConvertPathToFullPath(pwzArgs[0], pConfig->QueryApplicationPhysicalPath()->QueryStr(), &struTempPath)))
+    {
+        pwzArgs[0] = struTempPath.QueryStr();
+    }
+
```

Comment:
argv[0] = SysAllocString, you may want to do NULL check on argv[0] and argv[1] to see whether the SysAllocString success or not

Reply:
pwzArgs[0] will actually be argv[2]. I can add those checks.